WordPress/gutenberg
# [Mobile] - Try splitting E2E tests

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5455

## What?
This PR splits the only E2E test we have in this repo into three.

## Why?
In the past, we've had issues when incrementing the content of the `initial-HTML` file for testing. I can't remember the goal for this test but I think it's mostly to check all blocks render correctly, it could also be to stress test the editor by adding a lot of content but I don't think this is the case.

After the React Native upgrade I believe we've started to have more flakiness on the iOS tests, I think this is because the rendering of the block list is a bit different, where it starts rendering content as batches and before it blocked the UI until It finished rendering the blocks.

For the `gutenberg-editor-initial-html.test.js` on iOS, we were expecting to get the last block right after setting the content. On Android, we scroll down up to the very end to find the block.

## How?
First, we split the `gutenberg-editor-initial-html.test.js` into three for this repo consisting on:

- Check text blocks render correctly
- Check media blocks render correctly
- Check other blocks render correctly

All of those are in separate tests, for all of them, it would use the Android approach to scroll down to the end to get the `Add Paragraph block` placeholder. It's worth noting that for iOS sometimes scrolling down **won't be needed** as it would find the block right away.

There's a new helper `scrollAndReturnElementByAccessibilityId` which is similar to `androidScrollAndReturnElement` but it's for both platforms and it looks for an accessibility ID instead.

For this repo we will only run those tests, we are updating the filter to take into account tests that have `rendering` as part of the names.

The initial HTML test will only run on Gutenberg Mobile, the reasoning for this is that there we have more powerful machines, whereas here I think we are using the most basic ones limiting the performance of the simulators.

## Testing Instructions
CI Checks should pass on both Gutenberg and Gutenberg mobile. Try re-running the E2E tests to see if they succeed again.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
N/A
